source/DomainTheory/BasesAndContinuity/CompactBasis-SplitSupport.lagda

```diff
@@ -0,0 +1,357 @@
+                                                        (⊑ₛ-to-⊑ k)))
+         λ {refl → reflexivityₛ x , reflexivityₛ x}
+     image-is-small : is-small (image β')
+     image-is-small = sr β' (B , ≃-refl B) K-is-locally-small K-is-set
+
+ ι-⊑-is-small : (x : ⟨ 𝓓 ⟩) (c : K) → is-small (ι c ⊑⟨ 𝓓 ⟩ x)
+ ι-⊑-is-small x c = ∥∥-rec (being-small-is-prop ua (ι c ⊑⟨ 𝓓 ⟩ x) 𝓥)
+                           γ uscb
+  where
+   γ : has-specified-small-compact-basis 𝓓
+     → is-small (ι c ⊑⟨ 𝓓 ⟩ x)
+   γ (B , β , scb) =
+    ⌜ local-smallness-equivalent-definitions 𝓓 ⌝ loc-small (ι c) x
+     where
+      loc-small : is-locally-small 𝓓
+      loc-small = locally-small-if-small-compact-basis 𝓓 β scb
+
+\end{code}
+
+We then use a resized version of K and ι to get our small compact basis.
+
+The results about directedness and suprema of ↓ᴷ-inclusion apply almost directly
+as we are merely reindexing the family along an equivalence to a small type.
+
+\begin{code}
+
+ Kₛ : 𝓥 ̇
+ Kₛ = resized K K-is-small
+
+ ιₛ : Kₛ → ⟨ 𝓓 ⟩
+ ιₛ = ι ∘ ⌜ resizing-condition K-is-small ⌝
+
+ ↓-resizing : (x : ⟨ 𝓓 ⟩) → ↓ᴷ x ≃ ↓ᴮ 𝓓 ιₛ x
+ ↓-resizing x =
+  ≃-sym (Σ-change-of-variable _ ⌜ resizing-condition K-is-small ⌝
+                              (⌜⌝-is-equiv (resizing-condition K-is-small)))
+
+ ↓ᴷₛ-is-directed : (x : ⟨ 𝓓 ⟩) → is-Directed 𝓓 (↓-inclusion 𝓓 ιₛ x)
+ ↓ᴷₛ-is-directed x =
+  reindexed-family-is-directed 𝓓 (↓-resizing x) (↓ᴷ-inclusion x) (↓ᴷ-is-directed x)
+
+ ιₛ-is-small-compact-basis : is-small-compact-basis 𝓓 ιₛ
+ ιₛ-is-small-compact-basis =
+  record
+   { basis-is-compact = λ k → ι-is-compact (⌜ resizing-condition K-is-small ⌝ k)
+   ; ⊑ᴮ-is-small = λ x k → ι-⊑-is-small x (⌜ resizing-condition K-is-small ⌝ k)
+   ; ↓ᴮ-is-directed = ↓ᴷₛ-is-directed
+   ; ↓ᴮ-is-sup = λ x → reindexed-family-sup 𝓓 (↓-resizing x)
+                                              (↓ᴷ-inclusion x) x (↓ᴷ-is-sup x)
+   }
+
+\end{code}
+
+We summarise the main results of this file below.
+
+\begin{code}
+
+open import UF.KrausLemma
+open split-support-and-collapsibility pt
+
+module _
+        (ua : Univalence)
+        (sr : Set-Replacement pt)
+        (𝓓 : DCPO {𝓤} {𝓣})
+       where
+
+ specified-small-compact-basis-has-split-support :
+  has-split-support (has-specified-small-compact-basis 𝓓)
+ specified-small-compact-basis-has-split-support uscb =
+  Kₛ , ιₛ , ιₛ-is-small-compact-basis
+   where
+    open basis-of-compact-elements-preliminaries 𝓓 uscb
+    open basis-of-compact-elements ua sr 𝓓 uscb
+
+ specified-unspecified-equivalence :
+  has-specified-small-compact-basis 𝓓 ⇔ has-unspecified-small-compact-basis 𝓓
+ specified-unspecified-equivalence =
+    ∣_∣
+  , specified-small-compact-basis-has-split-support
+
+\end{code}
+
+We stress that the above cannot be promoted to an equivalence of types as there
+could be several small compact bases for 𝓓. In a sense, the above says that
+there is a canonical one though (which is given by all compact elements).
```

Comment:
Any basis of an algebraic dcpo has to include all compact elements. You probably mean something else.

Reply:
The definition of small compact basis allows for some redundancy, e.g. by taking two copies of all compact elements. In any case, I'll try to clarify the comment, but I need to think a bit about that. Thanks.

Reply:
Shouldn't there be a unique basis which is an embedding, even without requiring smallness conditions?

Reply:
Yes, that's right. In fact, I explain to include this as part of an improved explanation in a commit later today.
Once I'm done, I'll ping you here to get your thoughts on it. Is that all right?